twake/backend/node/src/services/notifications/services/engine/processors/reaction-notification.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  NotificationMessageQueueHandler,
+  PushNotificationMessage,
+  ReactionNotification,
+} from "../../../types";
+import { logger } from "../../../../../core/platform/framework";
+import { websocketEventBus } from "../../../../../core/platform/services/realtime/bus";
+import {
+  RealtimeEntityActionType,
+  ResourcePath,
+} from "../../../../../core/platform/services/realtime/types";
+import { MobilePushNotifier } from "../../../notifiers";
+import gr from "../../../../global-resolver";
+import { getNotificationRoomName } from "../../realtime";
+import { Channel } from "../../../../channels/entities/channel";
+import User from "../../../../user/entities/user";
+export class PushReactionNotification
+  implements NotificationMessageQueueHandler<ReactionNotification, void>
+{
+  readonly topics = {
+    in: "notification:reaction",
+  };
+
+  readonly options = {
+    unique: true,
+    ack: true,
+  };
+
+  name = "PushReactionToUsersMessageProcessor";
+
+  validate(message: ReactionNotification): boolean {
+    return !!(
+      message &&
+      message.message_id &&
+      message.channel_id &&
+      message.company_id &&
+      message.workspace_id &&
+      message.thread_id &&
+      message.user_id &&
+      message.reaction &&
+      message.creation_date
+    );
+  }
+
+  async process(message: ReactionNotification): Promise<void> {
+    logger.info(
+      `${this.name} - Processing reaction notification for message ${message.message_id}`,
+    );
+
+    if (
+      !message.company_id ||
+      !message.workspace_id ||
+      !message.channel_id ||
+      !message.message_id ||
+      !message.thread_id ||
+      !message.reaction ||
+      !message.user_id ||
+      !message.creation_date
+    ) {
+      throw new Error("Missing required fields");
```

Comment:
This is not needed because you do it in « validate » (which is called by the rabbitmq task processor)

Reply:
done